src/Compilers/Core/Portable/CodeGen/LocalDefinition.cs

```diff
@@ -56,8 +58,8 @@ internal sealed class LocalDefinition : Cci.ILocalDefinition
         /// <param name="dynamicTransformFlags">The synthesized dynamic attributes of the local.</param>
         /// <param name="tupleElementNames">Tuple element names of the local.</param>
         public LocalDefinition(
-            ILocalSymbol symbolOpt,
-            string nameOpt,
+            ILocalSymbol? symbolOpt,
```

Comment:
Part of me wonders if we should drop the Opt part of the name now we have annotations (probably not for this PR but just thinking aloud). 

It made sense before, but now that we can express the intent with the type system it seems superfluous to encode it into the name as well.

Reply:
We should, but I did not make these changes here to avoid unnecessary code churn. We can create an analyzer to aid in this polishing effort once more of the code base is annotated.

Reply:
Filed https://github.com/dotnet/roslyn-analyzers/issues/2920 for the analyzer